clippy_lints/src/utils/internal_lints/metadata_collector.rs

```diff
@@ -214,6 +258,107 @@ impl Serialize for ApplicabilityInfo {
     }
 }
 
+// ==================================================================
+// Configuration
+// ==================================================================
+#[derive(Debug)]
+pub(crate) struct ClippyConfigurationBasicInfo {
+    pub name: &'static str,
+    pub config_type: &'static str,
+    pub default: &'static str,
+    pub doc_comment: &'static str,
+    pub deprecation_reason: Option<&'static str>,
+}
+
+#[derive(Debug, Clone, Default)]
+struct ClippyConfiguration {
+    name: String,
+    lints: Vec<String>,
+    doc: String,
+    config_type: &'static str,
+    default: String,
+    deprecation_reason: Option<&'static str>,
+}
```

Comment:
Would it be simpler to skip `ClippyConfigurationBasicInfo` and just create `ClippyConfiguration` in the macro?

Reply:
The main reason why I have done it this way is to keep the monster contained to one module. It surely has its fangs in some other parts, but doing it this way keeps it mostly locked up.

But thinking about it, I could simply create a function in the module for it. *Sadly* a good suggestion as it requires some changes ^^